### PR TITLE
[Merged by Bors] - feat: add `Algebra ℝ A` instance given `Algebra ℂ A`

### PR DESCRIPTION
**Pull request by j-loreaux**
---

- [x] depends on: #9037

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Complex/Module.lean [hunk 1]
```diff
@@ -177,6 +177,18 @@ instance (priority := 900) Module.complexToReal (E : Type*) [AddCommGroup E] [Mo
   RestrictScalars.module ℝ ℂ E
 #align module.complex_to_real Module.complexToReal
 
+/- Register as an instance (with low priority) the fact that a complex algebra is also a real
+algebra. -/
+instance (priority := 900) Algebra.complexToReal {A : Type*} [Semiring A] [Algebra ℂ A] :
+    Algebra ℝ A :=
+  RestrictScalars.algebra ℝ ℂ A
+
+-- try to make sure we're not introducing diamonds.
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Module.lean:186]: Can you also check that the square with `Module.complexToReal` commutes?

File: Mathlib/Data/Complex/Module.lean [hunk 2]
```diff
@@ -177,6 +177,18 @@ instance (priority := 900) Module.complexToReal (E : Type*) [AddCommGroup E] [Mo
...
+example : Prod.algebra ℝ ℂ ℂ = (Prod.algebra ℂ ℂ ℂ).complexToReal := rfl
+example {ι : Type*} [Fintype ι] :
+    Pi.algebra (R := ℝ) ι (fun _ ↦ ℂ) = (Pi.algebra (R := ℂ) ι (fun _ ↦ ℂ)).complexToReal :=
+  rfl
+
 instance Module.real_complex_tower (E : Type*) [AddCommGroup E] [Module ℂ E] :
```
Comment (eric-wieser) [inline Mathlib/Data/Complex/Module.lean:192]: I think this instance is made pointless by your new one below; but I haven't *checked* that the latter implies the former.

Comment (j-loreaux) [reply Mathlib/Data/Complex/Module.lean:192]: indeed, it does, but I left an `example` of the old one to be sure.

File: Mathlib/Data/Complex/Module.lean (hunk 1 above)
Comment (j-loreaux) [reply Mathlib/Data/Complex/Module.lean:186]: done

**Comment by eric-wieser** [comment]
!bench

**Comment by eric-wieser** [comment]
Quoting a private message:


> Note: for #8991 this also works:
> ```lean
> import Mathlib
> 
> variable {A} [Ring A] [Algebra ℝ A]
> open Algebra TensorProduct
> 
> example : (leftAlgebra : Algebra ℂ (ℂ ⊗[ℝ] A)).complexToReal = leftAlgebra := rfl
> ```


I'd like to see this committed too. It can go in `test/instance_diamonds.lean` once #9037 goes in

**Comment by eric-wieser** [comment]
bors d+

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Pull request closed without merging**